src/transformers/models/switch/modeling_switch.py

```diff
@@ -381,10 +336,20 @@ def __init__(self, config: SwitchConfig):
         self.linear = nn.Linear(self.config.d_model, self.config.n_experts)
         self.softmax = nn.Softmax(dim=-1)
 
+    # Define getters, MUST call getters after forward
+    def get_expert_gate(self):
+        return self.expert_gate
+    def get_expert_mask(self):
+        return self.expert_mask
+    def get_router_probs(self):
+        return self.router_probs
+    def get_tokens_per_core(self):
```

Comment:
Why are these here? Why can't we just call self.tokens_per_core? If they must be called later, perhaps use the @property syntax.

Reply:
In our case, we have two separate classes, and because we want to pass the variable across classes, I am passing it that way. I'm not familiar of any elegant way to do it with Python, in C++ I normally would use inheritance or friend functions in scenarios like these. 

I did not use `@property` earlier because we just have getters and not setters. But I can add it for now.

Reply:
We have two separate classes only because of the way we calculate the losses